File: src/kinematics_rb1/velocity_controller.cpp
#include "kinematics_rb1/velocity_controller.hpp"

VelocityController::VelocityController(ros::NodeHandle &nh, const std::string &topic) : nh_(nh) {
  cmd_vel = nh_.advertise<geometry_msgs::Twist>(topic, 100);
}

void VelocityController::move(geometry_msgs::Twist tw) {
    cmd_vel.publish(tw);
}

File: scripts/velocity_controller.py
#! /usr/bin/env python

import rospy
from geometry_msgs.msg import Twist


class VelocityController():
    def __init__(self, topic):
        self.cmd_vel = rospy.Publisher(topic, Twist, queue_size=10)
        rospy.sleep(0.1)

    def move(self, linear_velocity=0.0, angular_velocity=0.0):
        msg = Twist()
        msg.linear.x = linear_velocity
        msg.angular.z = angular_velocity
        self.cmd_vel.publish(msg)
